### PR TITLE
State machine

### PR DESCRIPTION
**Pull request by ridhitbhura**
Merge state_machine branch to master branch

### CODE REVIEW

**Review by davidsw34 — COMMENTED** [review]
lgtm. But let's have @roninsharma25  review it too since there are changes to the JSON that the UI team would read from

**Review by roninsharma25 — APPROVED** [review]
LGTM!

File: Communications/Communications_Code/subscribe.c [hunk 1]
```diff
@@ -96,20 +96,20 @@ void writeJson(char* fileName, const sensor_info_t *msg, size_t size) {
...
-			fprintf(fp, "%f", msg->imu_magnetometer);
-			// fprintf(fp, "%f", 1);
-			fprintf(fp, "\n");
-					fprintf(fp, "},\n");
+	// Magnetometer
+	fprintf(fp, "\"magnetometer\" : ");
+	// Print value
+	fprintf(fp, "%f", msg->imu_magnetometer);
+	// fprintf(fp, "%f", 1);
+	fprintf(fp, "\n");
```
Comment (roninsharma25) [inline Communications/Communications_Code/subscribe.c:111]: test comment

File: Communications/Communications_Code/subscribe.c [hunk 2]
```diff
@@ -119,8 +119,8 @@ void writeJson(char* fileName, const sensor_info_t *msg, size_t size) {
 	// fprintf(fp, "%f", 5.6);
 	fprintf(fp, "\n },");
 
-			// Temperature
-			fprintf(fp, "\"Temperature\" : \n{");
```
Comment (roninsharma25) [inline Communications/Communications_Code/subscribe.c:123]: why''d you remove this line

**Review by roninsharma25 — CHANGES_REQUESTED** [review]

**Review by roninsharma25 — APPROVED** [review]

**Pull request merged**